libs/vkd3d/command.c

```diff
@@ -2079,6 +2109,202 @@ static void d3d12_command_list_clear_attachment_inline(struct d3d12_command_list
     }
 }
 
+static VkImageLayout dsv_plane_optimal_mask_to_layout(uint32_t mask, VkImageAspectFlags aspects)
+{
+    const VkImageAspectFlags ds_aspects = VK_IMAGE_ASPECT_DEPTH_BIT | VK_IMAGE_ASPECT_STENCIL_BIT;
+    static const VkImageLayout layouts[] = {
+            VK_IMAGE_LAYOUT_DEPTH_STENCIL_READ_ONLY_OPTIMAL,
+            VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_STENCIL_READ_ONLY_OPTIMAL,
+            VK_IMAGE_LAYOUT_DEPTH_READ_ONLY_STENCIL_ATTACHMENT_OPTIMAL,
+            VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL,
+    };
+
+    if ((aspects & ds_aspects) != ds_aspects)
```

Comment:
Is `aspects` supposed to be the aspect mask supported by the image format? This isn't super clear from the naming.

Reply:
Need to check and comment a bit more.

Reply:
Make code clearer and added comment.